### PR TITLE
Handle ConsumerGroup finalization for deleted topics

### PR DESCRIPTION
**Pull request by pierDipi**
When a topic is deleted before a ConsumerGroup, we should
not fail to finalize the ConsumerGroup when deleting the
Consumer group in Kafka.

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

**Comment by pierDipi** [comment]
/test unit-tests

Flake? 
`{Failed  === RUN   TestReconcileKind/Consumers_in_multiple_pods,_with_pods_pending_and_unknown_phase`
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/2816/unit-tests_eventing-kafka-broker_main/1597918389655834624

File: control-plane/pkg/reconciler/testing/errors.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testing
+
+// ErrorAssertOrNil returns an error given an interface{} variable
+func ErrorAssertOrNil(e interface{}) error {
```
Comment (matzew) [inline control-plane/pkg/reconciler/testing/errors.go:20]: is this something interesting for larger scope?

File: control-plane/pkg/reconciler/consumergroup/consumergroup.go [hunk 2]
```diff
@@ -187,12 +187,11 @@ func (r Reconciler) FinalizeKind(ctx context.Context, cg *kafkainternals.Consume
 	defer kafkaClusterAdminClient.Close()
 
 	groupId := cg.Spec.Template.Spec.Configs.Configs["group.id"]
-	if err := kafkaClusterAdminClient.DeleteConsumerGroup(groupId); err != nil && !errors.Is(sarama.ErrGroupIDNotFound, err) {
-		logging.FromContext(ctx).Errorw("unable to delete the consumer group", zap.String("id", groupId), zap.Error(err))
-		return err
+	if err = kafkaClusterAdminClient.DeleteConsumerGroup(groupId); err != nil && !errorIsOneOf(err, sarama.ErrUnknownTopicOrPartition, sarama.ErrGroupIDNotFound) {
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/consumergroup.go:190]: could we add a little comment that we give up, when actual kafka constructs are already gone?

File: control-plane/pkg/reconciler/testing/errors.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/testing/errors.go:20]: Maybe

File: control-plane/pkg/reconciler/consumergroup/consumergroup.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/consumergroup.go:190]: wouldn't that comment just restate what the code is already stating?

Comment (matzew) [reply control-plane/pkg/reconciler/consumergroup/consumergroup.go:190]: yeah... sorta

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**